src/commands.py

```diff
@@ -339,3 +346,222 @@ async def action(self, message):
     @staticmethod
     def helptxt():
         return "$help [command] \nDisplays description of the provided command. If none is provided, displays description for all commands."
+
+class menu(command_on_message):
+    def __init__(self, *args, **kwargs):
+        self.commons = ['dlg', 'de-la-guerra', 'portola', 'carrillo', 'ortega'] # 'de' is added in case someone searches for 'de la guerra' instead of 'dlg'
+        self.mealtimes = ['breakfast', 'lunch', 'dinner', 'brunch']
+        self.day = 0 # Force a menu update due to get_menu being asynchronous.
+        self.updating_menus = False
+        super().__init__(*args, **kwargs)
+
+    async def action(self, message: Message):
+        '''
+        Roadmap Idea - Daniel. I've implemented code and pseudo-code for this idea below. This is not final by any means.
+
+        First, check menu. If it is outdated, then update it.
+        Second, create some base states and all the actions.
+        Third, parse the user's information
+        Fourth, using the parsed information, put the machine into the right initial state.
+        '''
+        # To prevent multiple calls to API, cancel early with a "try again". Is it possible to wait for a value to be changed though?
+        if self.updating_menus: return message_data(channel=message.channel, message="Please try again momentarily, as the menu is currently being updated.")
+
+        message_to_replace = None
+        current_date = date.today()
+        if current_date.day != self.day: # If the days fail to align, update the menu.
+            message_to_replace = await message.channel.send('Fetching the menus...')
+            self.updating_menus = True
+            self.menu, self.day = await get_allmenusaio(current_date.isoformat()), current_date.day
```

Comment:
I don't see `self.menu` as part of `__init__` or a @property. Let's make it explicit to avoid having to dig through code. I would suggest something like `self._menu = None` in `__init__` and making `self.menu` a @property that raises an error if `self._menu` is None.